### PR TITLE
feat: add webhooks support

### PR DESCRIPTION
**Pull request by stasiukanya**
This PR implements webhooks support.

**Example of usage:**

```yaml
x-webhooks:
  newPet:
    post:
      summary: New pet
      description: Information about a new pet in the systems
      operationId: newPet
      tags: 
        - pet
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/Pet"
      responses:
        "200":
          description: Return a 200 status to indicate that the data was received successfully
```

**Layout:**

image

### CODE REVIEW

File: src/components/SideMenu/MenuItem.tsx [hunk 1]
```diff
@@ -90,7 +90,11 @@ export class OperationMenuItemContent extends React.Component<OperationMenuItemC
         deprecated={item.deprecated}
         ref={this.ref}
       >
-        <OperationBadge type={item.httpVerb}>{shortenHTTPVerb(item.httpVerb)}</OperationBadge>
+        {item.isWebhook ? (
+          <OperationBadge type="hook">hook</OperationBadge>
```
Comment (adamaltman) [inline src/components/SideMenu/MenuItem.tsx:94]: I wonder why not `webhook`?

Comment (RomanHotsiy) [reply src/components/SideMenu/MenuItem.tsx:94]: Check out the screenshot above.`webhook` will be wider than the rest badges and will look ugly.

File: src/services/SpecStore.ts [hunk 2]
```diff
@@ -15,6 +16,7 @@ export class SpecStore {
   externalDocs?: OpenAPIExternalDocumentation;
   contentItems: ContentItemModel[];
   securitySchemes: SecuritySchemesModel;
+  'x-webhooks'?: WebhookModel;
```
Comment (okachynskyy) [inline src/services/SpecStore.ts:19]: What about `xWebhooks` or `xWebHooks`?

Comment (RomanHotsiy) [reply src/services/SpecStore.ts:19]: This is a convention by OpenAPI spec, all vendor extensions should prefixed by `x-`.

Comment (RomanHotsiy) [reply src/services/SpecStore.ts:19]: Oops, my bad. This is our code not typing for OAS. @stasiukanya could you update it to `xWebhooks`? And everywhere else.

File: src/services/SpecStore.ts [hunk 3]
```diff
@@ -26,5 +28,6 @@ export class SpecStore {
     this.externalDocs = this.parser.spec.externalDocs;
     this.contentItems = MenuBuilder.buildStructure(this.parser, this.options);
     this.securitySchemes = new SecuritySchemesModel(this.parser);
+    this['x-webhooks'] = new WebhookModel(this.parser, options, this.parser.spec['x-webhooks']);
```
Comment (RomanHotsiy) [inline src/services/SpecStore.ts:31]:
```suggestion
    this.xWebhooks = new WebhookModel(this.parser, options, this.parser.spec['x-webhooks']);
```

File: src/services/SpecStore.ts (hunk 2 above)
Comment (adamaltman) [reply src/services/SpecStore.ts:19]: Why not `webhooks` then if it is our code?

File: src/components/SideMenu/MenuItem.tsx (hunk 1 above)
Comment (adamaltman) [reply src/components/SideMenu/MenuItem.tsx:94]: This label/badge needs to be configurable. "hook" is a pejorative word in various vernacular.

File: src/services/SpecStore.ts (hunk 2 above)
Comment (RomanHotsiy) [reply src/services/SpecStore.ts:19]: @adamaltman good catch!

File: src/components/SideMenu/MenuItem.tsx (hunk 1 above)
Comment (RomanHotsiy) [reply src/components/SideMenu/MenuItem.tsx:94]: @adamaltman Do you have any other ideas? Maybe `whook` or `webh`? We use `opts` for `options` and `del` for `delete`.

**Comment by RomanHotsiy** [comment]
🎉 👏

**Comment by RomanHotsiy** [comment]
Hey @lornajane! We've already released x-webhooks support in the latest 2.0.0-rc.37 and redoc-cli@0.9.11.

Could you take a look and let us know what you think? We still haven't announced it, wanted to get your feedback before!

**Comment by lornajane** [comment]
I love this! The feature looks exactly what I expected (basically callbacks, but in a different context), and feels familiar. I think this is the catalyst I needed to go ahead and start updating our specs within the 3.0 versions to include this, ready for an official OpenAPI 3.1 release!

**Pull request merged**